File: src/_058/kotlin/Solution.kt
package _058.kotlin

/**
 * @author relish
 * @since 2018/04/18
 */
class Solution {
    fun lengthOfLastWord(s: String): Int {
        var count = 0
        for (i in s.length - 1 downTo 0) {
            if (s[i] == ' ') {
                if (count == 0) continue
                return count
            } else {
                count++
            }
        }
        return count
    }
}

fun main(args: Array<String>) {
    println(Solution().lengthOfLastWord("a "))
    println(Solution().lengthOfLastWord("Hello World"))
}

File: src/_058/Solution.js
var lengthOfLastWord = function(s) {
    var len = s.length - 1;
    for (var i = len; i >= 0; i--) {
        if(s[i] == ' ') {
            len--
        } else {
            break;
        }
    }
    var realLen = len
    for (var i = len; i >= 0; i--) {
        if(s[i] == ' ') {
            break;
        } else {
            len--
        }
    }
	return realLen - len
};
var lengthOfLastWord = function(s) {
    var len = s.length - 1;
    while (len >= 0 && s[len] == ' ') len--;
    var realLen = len;
    while (len >= 0 && s[len] != ' ') len--;
    return realLen - len;
};
